assets/src/settings-page/analytics.js

```diff
@@ -111,11 +112,16 @@ function AnalyticsEntry( { entryIndex, onChange, onDelete, type = '', config = '
 
 	const defaultValue = vendorConfigs[ type ]?.sample || '{}';
 
-	/** @type {string} value */
-	const value = ( '' === config.trim() || Object.values( vendorConfigs ).find( ( { sample } ) => sample === config ) ) && ! isTextareaFocused()
+	/** @type {string} textareaControlValue */
+	const textareaControlValue = ( '' === config.trim() || Object.values( vendorConfigs ).find( ( { sample } ) => sample === config ) ) && ! isTextareaFocused()
 		? defaultValue
 		: config;
 
+	/* Populating options for analytics vendors. */
+	const options = [
+		...Object.entries( AnalyticsVendors ).map( ( [ value, label ] ) => ( { value, label } ) ),
+	];
+
```

Comment:
Two things about this part:

1. There's no need to use an array spread now. `Object.entries()` already returns an array and we're just `map`ping over it so the resulting value is an array all of the time. (Earlier, since we've been prepending an empty option to the `options` array, we had to use an array spread.) 

```suggestion
	/* Populating options for analytics vendors. */
	const options = Object.entries( AnalyticsVendors ).map( ( [ value, label ] ) => ( { value, label } ) );

```

2. This code should be moved outside of the functional component, i.e. outside of the `AnalyticsEntry` so that we're not recreating the `options` array on each component render. The `AnalyticsVendors` object is "static" (it's imported from a JSON file) so it doesn't depend on a component state and so we can do this mapping only once.

Reply:
Resolved the issue.